### PR TITLE
Cleaning ExprDirection patterns

### PR DESCRIPTION
**Pull request by Fusezion**
### Description

This PR aims to try and cleanup some of the mess in ExprDirection more notably the patterns where it was basically unreadable to anyone regardless how much you try to split it up. I intended to cause no breaking changes however if a breaking change appears please let me know so I can change it 

The biggest change is gonna be where `north`, `east`, `south`, `west`, `up`, `down` pattern was. They been split up into
general direction and up/down. This should provide an easier time in reading while also making the most of parse tags.

Now I'm sure there's gonna be a bit of concern about the enum and "tags" field used however as I stated in the comment I saw no other workaround for this, I'm already using a mildly janky pattern to support every direction without causing any breaking changes. All while trying not to use parse marks just to save source code readability couldn't stand seeing `blah ^ blah` everywhere just for some basic `north west`, `north east` support

I would however love some feedback on a better enum name as "DirectionMapping" was simply a placeholder or suppose to be.

## Pattern Before/After
*these were edited to the way they appear on the docs as such bits might of been accidently removed*
I didn't realize `[to the]` was apart of the `[%number ....]` until I added this however since `to the south` makes sense I have no intention of moving it back

### Before
```applescript
[%number% [(block|met(er|re))[s]] [to the]] (north[[(-| )](east|west)][(ward[(s|ly)]|er[(n|ly)])] [of]|south[[(-| )](east|west)][(ward[(s|ly)]|er[(n|ly)])] [of]|(east|west)[(ward[(s|ly)]|er[(n|ly)])] [of]|above|over|(up|down)[ward[(s|ly)]]|below|under[neath]|beneath) [%direction%]
[%number% [(block|met(er|re))[s]]] in [the] (direction|horizontal direction|facing|horizontal facing) of %entity/block% [(of|from)]
[%number% [(block|met(er|re))[s]]] in %entity/block%'[s] (direction|horizontal direction|facing|horizontal facing) [(of|from)]
[%number% [(block|met(er|re))[s]]] (in[ ]front [of]|forward[s]|behind|backwards|[to the] (right|left) [of])
[%number% [(block|met(er|re))[s]]] horizontal[ly] (in[ ]front [of]|forward[s]|behind|backwards|to the (right|left) [of])
```
### After
```applescript
[%number% [(block|met(er|re))[s]]] [to the] (:(north[[-| ](east|west)]|east|south[[ |-](east|west)]|west))[ward[s|ly]|er(n|ly)] [of] [%direction%]
[%number% [(block|met(er|re))[s]]] [to the] (up:(above|over|up[ward[s|ly]])|down:(down[ward[s|ly]]|below|under[neath]|beneath)) [%direction%]
[%number% [(block|met(er|re))[s]]] in [the] [horizontal] (direction|facing) of %entity/block% [of|from]
[%number% [(block|met(er|re))[s]]] in %entity/block%'[s] [horizontal] (direction|facing) [of|from]
[%number% [(block|met(er|re))[s]]] [horizontal[ly]] (in[ ]front [of]|forward[s]|(behind|backwards)|[to the] (right|left) [of])
```

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

**Comment by Fusezion** [comment]
build seems to be failing on the `3281-itemamount fixes.sk` regression test
not sure why tho

File: src/main/java/ch/njol/skript/expressions/ExprDirection.java [hunk 1]
```diff
@@ -60,153 +60,181 @@
...
-						"|(" + EAST + "¦east|" + WEST + "¦west)[(ward(s|ly|)|er(n|ly|))] [of]" +
-						"|" + UP + "¦above|" + UP + "¦over|(" + UP + "¦up|" + DOWN + "¦down)[ward(s|ly|)]|" + DOWN + "¦below|" + DOWN + "¦under[neath]|" + DOWN + "¦beneath" +
-						") [%-direction%]",
-				"[%-number% [(block|met(er|re))[s]]] in [the] (0¦direction|1¦horizontal direction|2¦facing|3¦horizontal facing) of %entity/block% (of|from|)",
-				"[%-number% [(block|met(er|re))[s]]] in %entity/block%'[s] (0¦direction|1¦horizontal direction|2¦facing|3¦horizontal facing) (of|from|)",
-				"[%-number% [(block|met(er|re))[s]]] (0¦in[ ]front [of]|0¦forward[s]|2¦behind|2¦backwards|[to the] (1¦right|-1¦left) [of])",
-				"[%-number% [(block|met(er|re))[s]]] horizontal[ly] (0¦in[ ]front [of]|0¦forward[s]|2¦behind|2¦backwards|to the (1¦right|-1¦left) [of])");
+				"[%-number% [(block|met(er|re))[s]]] [to the] (:(north[[-| ](east|west)]|east|south[[ |-](east|west)]|west))[ward[s|ly]|er(n|ly)] [of] [%-direction%]",
+				"[%-number% [(block|met(er|re))[s]]] [to the] (up:(above|over|up[ward[s|ly]])|down:(down[ward[s|ly]]|below|under[neath]|beneath)) [%-direction%]",
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/expressions/ExprDirection.java:111]: I don't think tags work like you think they do. This should only ever have the tags `north`, `west`, `south`, `east`, `up`, or `down`

2.8.4:
```
> !send north
[21:18:22 INFO]: executing 'send north'
[21:18:22 INFO]: north
> !send north-west
[21:18:24 INFO]: executing 'send north-west'
[21:18:24 INFO]: west and north
> !send east
[21:18:28 INFO]: executing 'send east'
[21:18:28 INFO]: east
> !send south
[21:18:30 INFO]: executing 'send south'
[21:18:30 INFO]: south
> !send north-east
[21:18:33 INFO]: executing 'send north-east'
[21:18:33 INFO]: east and north
```

nightly:
```
> !send north-east
[21:15:12 INFO]: executing 'send north-east'
[21:15:12 INFO]: north
> !send south
[21:15:19 INFO]: executing 'send south'
[21:15:19 INFO]: south
> !send east
[21:15:22 INFO]: executing 'send east'
[21:15:22 INFO]: east
> !send north-west
[21:15:28 INFO]: executing 'send north-west'
[21:15:28 INFO]: north
```

this pr also needs tests!

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprDirection.java (hunk 1 above)
Comment (sovdeeth) [reply src/main/java/ch/njol/skript/expressions/ExprDirection.java:111]: Oh, I also think that this isn't any more readable, tbh
It should really be split into a lot more patterns if you want to truly make it readable.

Comment (Fusezion) [reply src/main/java/ch/njol/skript/expressions/ExprDirection.java:111]: Alrighty then something I did in the past was very wrong and I'll work on fixing this converted to draft and should be able to finish this after work I do leave now but should be back in the next 6-ish hours shouldn't be long after that

**Comment by Fusezion** [comment]
closing for rn will open with a new one rebasing just went to utter crap and I shouldn't of done it

**Pull request closed without merging**